widgetastic_manageiq/__init__.py

```diff
@@ -1855,29 +1855,290 @@ def is_displayed(self):
+        self.apply_filter()
+        self._process_user_filling(user_input, cancel_on_user_filling)
+        self.close_advanced_search()
+
+    def _process_user_filling(self, user_input, cancel_on_user_filling=False):
+        """ This function handles answering CFME's requests on user input.
+
+        A `user_input` function is passed. If the box with user input appears, all requested
+        inputs are gathered and iterated over. On each element the `user_input` function is
+        called
+        with 2 parameters: text which precedes the element itself to do matching, and the element.
+
+        This function does not check return status after `user_input` call.
+
+            Args:
+                user_input: The function to be called on each user input.
+        """
+        user_input_form = self.advanced_search_form.filter_user_input_form
+        wait_for(
+            lambda: self.advanced_search_form.load_filter_button.is_displayed, fail_condition=True,
+            num_sec=10, delay=2, message='Waiting for button became active')
+        if isinstance(user_input, dict):
+            for user_input_label, user_input_value in user_input.items():
+                field_for_input = self.browser.element(
+                    user_input_form.USER_INPUT_FIELD.format(quote(user_input_label)))
+                field_for_input.send_keys(user_input_value)
+            if cancel_on_user_filling:
+                user_input_form.user_input_cancel.click()
+            else:
+                wait_for(
+                    lambda: user_input_form.user_input_apply.is_displayed,
+                    num_sec=10, delay=2, message='Waiting for button became active')
+                user_input_form.user_input_apply.click()
+
+    @property
+    def is_advanced_search_possible(self):
+        """Checks for advanced search possibility in the quadicon view"""
+        return self.advanced_search_button.is_displayed
+
+    @property
+    def is_advanced_search_applied(self):
+        """Checks whether any filter is in effect on quadicon view"""
+        return self.filter_clear_button.is_displayed
+
+    def open_advanced_search(self):
+        """Make sure the advanced search box is opened. """
+        if not self.is_advanced_search_opened:
+            self.advanced_search_button.click()
+            wait_for(lambda: self.is_advanced_search_opened, fail_condition=False,
+                     num_sec=10, delay=2, message='Waiting for advanced search to open')
+
+    def close_advanced_search(self):
+        """Checks if the advanced search box is open and if it does, closes it."""
+        if self.is_advanced_search_opened:
+            self.advanced_search_form.close_button.click()
+            wait_for(lambda: self.is_advanced_search_opened, fail_condition=True,
+                     num_sec=10, delay=2, message='Waiting for advanced search to close')
+
+    def remove_search_filters(self):
```

Comment:
REQUIRED: Please fix usages of search in widgetastic_manageiq

Reply:
I think  I've resolved this with:
https://github.com/ManageIQ/integration_tests/pull/5971/commits/dbf2391899f8bc8f5f4f33d4d24cb91f65136dfa